config/v1/types_cluster_monitoring.go

```diff
@@ -0,0 +1,88 @@
+// ClusterMonitoring is the Schema for the Cluster Monitoring Operators API
+type ClusterMonitoring struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user configuration for the Cluster Monitoring Operator
+	// +kubebuilder:validation:Required
+	Spec ClusterMonitoringSpec `json:"spec,omitempty"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterMonitoringStatus `json:"status,omitempty"`
+}
+
+// MonitoringOperatorStatus defines the observed state of MonitoringOperator
+type ClusterMonitoringStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:internal
+type ClusterMonitoringList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ClusterMonitoring `json:"items"`
+}
+
+// ClusterMonitoringSpec defines the desired state of Cluster Monitoring Operator
+type ClusterMonitoringSpec struct {
+	// userDefined set the deployment mode for user-defined monitoring in addition to the default platform monitoring.
+	UserDefined UserDefinedMonitoring `json:"userDefined,omitempty"`
+}
+
+// UserDefinedMonitoring config for user-defined projects.
+type UserDefinedMonitoring struct {
+	// mode defines the different configurations of UserDefinedMonitoring
+	// Valid values are UserDefinedDisabled and UserDefinedNamespaceIsolation
+	// UserDefinedDisabled disables monitoring for user-defined projects. This restricts the default monitoring stack, installed in the openshift-monitoring project, to monitor only platform namespaces, which prevents any custom monitoring configurations or resources from being applied to user-defined namespaces.
+	// UserDefinedNamespaceIsolation enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
+	// +kubebuilder:validation:Enum:="Disabled";"NamespaceIsolation"
+	Mode UserDefinedMode `json:"mode,omitempty"`
+}
+
+type UserDefinedMode string
+
+const (
+	// UserDefinedDisabled disables monitoring for user-defined projects. This restricts the default monitoring stack, installed in the openshift-monitoring project, to monitor only platform namespaces, which prevents any custom monitoring configurations or resources from being applied to user-defined namespaces.
+	UserDefinedDisabled UserDefinedMode = "Disabled"
+	// UserDefinedNamespaceIsolation enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
+	UserDefinedNamespaceIsolation UserDefinedMode = "NamespaceIsolation"
```

Comment:
Should this really be `NamespaceIsolated`?

Reply:
I don't feel strongly about this but I'd argue the current `NamespaceIsolation` is better. Its not the namespace that is isloated but the metrics access that is isolated by namespace values.

Reply:
> Its not the namespace that is isloated but the metrics access that is isolated by namespace values.

You've used the word `isolated` to describe how the metrics are accessed here, and I think to you're point, if it were `IsolatedNamespace`, I'd agree that this would imply the namespace is isolated, but my suggestions was `NamespaceIsolated`

I'm fine with either, but my thought was that `NamespaceIsolated` felt more declarative, and inline with `Disabled`. Isolation is a process, Isolated is a state, and it's the state that declarative APIs are supposed to represent

Reply:
Yeah fair point, I'm convinced. We also use `isolated at the namespace level` in the doc comments.

Reply:
yeah, it'll probably make more sense to be coherent all along the doc comments and code.